// @learncard/types@5.12.3/dist/types.esm.js
var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });

// src/vc.ts
import { z } from "zod";
var ContextValidator = z.array(z.string().or(z.record(z.string(), z.any())));
var AchievementCriteriaValidator = z.object({
  type: z.string().optional(),
  narrative: z.string().optional()
});
var ImageValidator = z.string().or(
  z.object({
    id: z.string(),
    type: z.string(),
    caption: z.string().optional()
  })
);
var GeoCoordinatesValidator = z.object({
  type: z.string().min(1).or(z.string().array().nonempty()),
  latitude: z.number(),
  longitude: z.number()
});
var AddressValidator = z.object({
  type: z.string().min(1).or(z.string().array().nonempty()),
  addressCountry: z.string().optional(),
  addressCountryCode: z.string().optional(),
  addressRegion: z.string().optional(),
  addressLocality: z.string().optional(),
  streetAddress: z.string().optional(),
  postOfficeBoxNumber: z.string().optional(),
  postalCode: z.string().optional(),
  geo: GeoCoordinatesValidator.optional()
});
var IdentifierTypeValidator = z.enum([
  "sourcedId",
  "systemId",
  "productId",
  "userName",
  "accountId",
  "emailAddress",
  "nationalIdentityNumber",
  "isbn",
  "issn",
  "lisSourcedId",
  "oneRosterSourcedId",
  "sisSourcedId",
  "ltiContextId",
  "ltiDeploymentId",
  "ltiToolId",
  "ltiPlatformId",
  "ltiUserId",
  "identifier"
]).or(z.string());
var IdentifierEntryValidator = z.object({
  type: z.string().min(1).or(z.string().array().nonempty()),
  identifier: z.string(),
  identifierType: IdentifierTypeValidator
});
var ProfileValidator = z.string().or(
  z.object({
    id: z.string().optional(),
    type: z.string().or(z.string().array().nonempty().optional()),
    name: z.string().optional(),
    url: z.string().optional(),
    phone: z.string().optional(),
    description: z.string().optional(),
    endorsement: z.any().array().optional(),
    // Recursive type
    image: ImageValidator.optional(),
    email: z.string().email().optional(),
    address: AddressValidator.optional(),
    otherIdentifier: IdentifierEntryValidator.array().optional(),
    official: z.string().optional(),
    parentOrg: z.any().optional(),
    // Recursive types are annoying =(
    familyName: z.string().optional(),
    givenName: z.string().optional(),
    additionalName: z.string().optional(),
    patronymicName: z.string().optional(),
    honorificPrefix: z.string().optional(),
    honorificSuffix: z.string().optional(),
    familyNamePrefix: z.string().optional(),
    dateOfBirth: z.string().optional()
  }).catchall(z.any())
);
var CredentialSubjectValidator = z.object({ id: z.string().optional() }).catchall(z.any());
var CredentialStatusValidator = z.object({ type: z.string(), id: z.string() }).catchall(z.any());
var CredentialSchemaValidator = z.object({ id: z.string(), type: z.string() }).catchall(z.any());
var RefreshServiceValidator = z.object({ id: z.string().optional(), type: z.string() }).catchall(z.any());
var TermsOfUseValidator = z.object({ type: z.string(), id: z.string().optional() }).catchall(z.any());
var VC2EvidenceValidator = z.object({
  id: z.string().optional(),
  type: z.array(z.string()).nonempty(),
  name: z.string().optional(),
  narrative: z.string().optional(),
  description: z.string().optional(),
  genre: z.string().optional(),
  audience: z.string().optional()
}).catchall(z.any());
var UnsignedVCValidator = z.object({
  "@context": ContextValidator,
  id: z.string().optional(),
  type: z.string().array().nonempty(),
  issuer: ProfileValidator,
  credentialSubject: CredentialSubjectValidator.or(CredentialSubjectValidator.array()),
  refreshService: RefreshServiceValidator.or(RefreshServiceValidator.array()).optional(),
  credentialSchema: CredentialSchemaValidator.or(
    CredentialSchemaValidator.array()
  ).optional(),
  // V1
  issuanceDate: z.string().optional(),
  expirationDate: z.string().optional(),
  credentialStatus: CredentialStatusValidator.or(
    CredentialStatusValidator.array()
  ).optional(),
  // V2
  name: z.string().optional(),
  description: z.string().optional(),
  validFrom: z.string().optional(),
  validUntil: z.string().optional(),
  status: CredentialStatusValidator.or(CredentialStatusValidator.array()).optional(),
  termsOfUse: TermsOfUseValidator.or(TermsOfUseValidator.array()).optional(),
  evidence: z.union([VC2EvidenceValidator, z.array(VC2EvidenceValidator)]).optional()
}).catchall(z.any());
var ProofValidator = z.object({
  type: z.string(),
  created: z.string(),
  challenge: z.string().optional(),
  domain: z.string().optional(),
  nonce: z.string().optional(),
  proofPurpose: z.string(),
  verificationMethod: z.string(),
  jws: z.string().optional()
}).catchall(z.any());
var VCValidator = UnsignedVCValidator.extend({
  proof: ProofValidator.or(ProofValidator.array())
});
var UnsignedVPValidator = z.object({
  "@context": ContextValidator,
  id: z.string().optional(),
  type: z.string().or(z.string().array().nonempty()),
  verifiableCredential: VCValidator.or(VCValidator.array()).optional(),
  holder: z.string().optional()
}).catchall(z.any());
var VPValidator = UnsignedVPValidator.extend({
  proof: ProofValidator.or(ProofValidator.array())
});

// src/did.ts
import { z as z3 } from "zod";

// src/crypto.ts
import { z as z2 } from "zod";
var JWKValidator = z2.object({
  kty: z2.string(),
  crv: z2.string(),
  x: z2.string(),
  y: z2.string().optional(),
  n: z2.string().optional(),
  d: z2.string().optional()
});
var JWKWithPrivateKeyValidator = JWKValidator.omit({ d: true }).extend({ d: z2.string() });
var JWERecipientHeaderValidator = z2.object({
  alg: z2.string(),
  iv: z2.string(),
  tag: z2.string(),
  epk: JWKValidator.partial().optional(),
  kid: z2.string().optional(),
  apv: z2.string().optional(),
  apu: z2.string().optional()
});
var JWERecipientValidator = z2.object({
  header: JWERecipientHeaderValidator,
  encrypted_key: z2.string()
});
var JWEValidator = z2.object({
  protected: z2.string(),
  iv: z2.string(),
  ciphertext: z2.string(),
  tag: z2.string(),
  aad: z2.string().optional(),
  recipients: JWERecipientValidator.array().optional()
});

// src/did.ts
var VerificationMethodValidator = z3.string().or(
  z3.object({
    "@context": ContextValidator.optional(),
    id: z3.string(),
    type: z3.string(),
    controller: z3.string(),
    publicKeyJwk: JWKValidator.optional(),
    publicKeyBase58: z3.string().optional(),
    publicKeyMultibase: z3.string().optional(),
    blockChainAccountId: z3.string().optional()
  }).catchall(z3.any())
);
var ServiceValidator = z3.object({
  id: z3.string(),
  type: z3.string().or(z3.string().array().nonempty()),
  serviceEndpoint: z3.any().or(z3.any().array().nonempty())
}).catchall(z3.any());
var DidDocumentValidator = z3.object({
  "@context": ContextValidator,
  id: z3.string(),
  alsoKnownAs: z3.string().optional(),
  controller: z3.string().or(z3.string().array().nonempty()).optional(),
  verificationMethod: VerificationMethodValidator.array().optional(),
  authentication: VerificationMethodValidator.array().optional(),
  assertionMethod: VerificationMethodValidator.array().optional(),
  keyAgreement: VerificationMethodValidator.array().optional(),
  capabilityInvocation: VerificationMethodValidator.array().optional(),
  capabilityDelegation: VerificationMethodValidator.array().optional(),
  publicKey: VerificationMethodValidator.array().optional(),
  service: ServiceValidator.array().optional(),
  proof: ProofValidator.or(ProofValidator.array()).optional()
}).catchall(z3.any());

// src/obv3.ts
import { z as z4 } from "zod";
var AlignmentTargetTypeValidator = z4.enum([
  "ceasn:Competency",
  "ceterms:Credential",
  "CFItem",
  "CFRubric",
  "CFRubricCriterion",
  "CFRubricCriterionLevel",
  "CTDL"
]).or(z4.string());
var AlignmentValidator = z4.object({
  type: z4.string().array().nonempty(),
  targetCode: z4.string().optional(),
  targetDescription: z4.string().optional(),
  targetName: z4.string(),
  targetFramework: z4.string().optional(),
  targetType: AlignmentTargetTypeValidator.optional(),
  targetUrl: z4.string()
});
var KnownAchievementTypeValidator = z4.enum([
  "Achievement",
  "ApprenticeshipCertificate",
  "Assessment",
  "Assignment",
  "AssociateDegree",
  "Award",
  "Badge",
  "BachelorDegree",
  "Certificate",
  "CertificateOfCompletion",
  "Certification",
  "CommunityService",
  "Competency",
  "Course",
  "CoCurricular",
  "Degree",
  "Diploma",
  "DoctoralDegree",
  "Fieldwork",
  "GeneralEducationDevelopment",
  "JourneymanCertificate",
  "LearningProgram",
  "License",
  "Membership",
  "ProfessionalDoctorate",
  "QualityAssuranceCredential",
  "MasterCertificate",
  "MasterDegree",
  "MicroCredential",
  "ResearchDoctorate",
  "SecondarySchoolDiploma"
]);
var AchievementTypeValidator = KnownAchievementTypeValidator.or(z4.string());
var CriteriaValidator = z4.object({ id: z4.string().optional(), narrative: z4.string().optional() }).catchall(z4.any());
var EndorsementSubjectValidator = z4.object({
  id: z4.string(),
  type: z4.string().array().nonempty(),
  endorsementComment: z4.string().optional()
});
var EndorsementCredentialValidator = UnsignedVCValidator.extend({
  credentialSubject: EndorsementSubjectValidator,
  proof: ProofValidator.or(ProofValidator.array()).optional()
});
var RelatedValidator = z4.object({
  id: z4.string(),
  "@language": z4.string().optional(),
  version: z4.string().optional()
});
var ResultTypeValidator = z4.enum([
  "GradePointAverage",
  "LetterGrade",
  "Percent",
  "PerformanceLevel",
  "PredictedScore",
  "RawScore",
  "Result",
  "RubricCriterion",
  "RubricCriterionLevel",
  "RubricScore",
  "ScaledScore",
  "Status"
]).or(z4.string());
var RubricCriterionValidator = z4.object({
  id: z4.string(),
  type: z4.string().array().nonempty(),
  alignment: AlignmentValidator.array().optional(),
  description: z4.string().optional(),
  level: z4.string().optional(),
  name: z4.string(),
  points: z4.string().optional()
}).catchall(z4.any());
var ResultDescriptionValidator = z4.object({
  id: z4.string(),
  type: z4.string().array().nonempty(),
  alignment: AlignmentValidator.array().optional(),
  allowedValue: z4.string().array().optional(),
  name: z4.string(),
  requiredLevel: z4.string().optional(),
  requiredValue: z4.string().optional(),
  resultType: ResultTypeValidator,
  rubricCriterionLevel: RubricCriterionValidator.array().optional(),
  valueMax: z4.string().optional(),
  valueMin: z4.string().optional()
}).catchall(z4.any());
var EvidenceValidator = z4.object({
  id: z4.string().optional(),
  type: z4.array(z4.string()).nonempty(),
  name: z4.string().optional(),
  narrative: z4.string().optional(),
  description: z4.string().optional(),
  genre: z4.string().optional(),
  audience: z4.string().optional()
}).catchall(z4.any());
var AchievementValidator = z4.object({
  id: z4.string().optional(),
  type: z4.array(z4.string()).nonempty(),
  alignment: AlignmentValidator.array().optional(),
  achievementType: AchievementTypeValidator.optional(),
  creator: ProfileValidator.optional(),
  creditsAvailable: z4.number().optional(),
  criteria: CriteriaValidator,
  description: z4.string(),
  endorsement: EndorsementCredentialValidator.array().optional(),
  fieldOfStudy: z4.string().optional(),
  humanCode: z4.string().optional(),
  image: ImageValidator.optional(),
  "@language": z4.string().optional(),
  name: z4.string(),
  otherIdentifier: IdentifierEntryValidator.array().optional(),
  related: RelatedValidator.array().optional(),
  resultDescription: ResultDescriptionValidator.array().optional(),
  specialization: z4.string().optional(),
  tag: z4.array(z4.string()).optional(),
  version: z4.string().optional()
}).catchall(z4.any());
var IdentityObjectValidator = z4.object({
  type: z4.string(),
  hashed: z4.boolean(),
  identityHash: z4.string(),
  identityType: z4.string(),
  salt: z4.string().optional()
});
var ResultStatusTypeValidator = z4.enum([
  "Completed",
  "Enrolled",
  "Failed",
  "InProgress",
  "OnHold",
  "Withdrew"
]);
var ResultValidator = z4.object({
  type: z4.string().array().nonempty(),
  achievedLevel: z4.string().optional(),
  alignment: AlignmentValidator.array().optional(),
  resultDescription: z4.string().optional(),
  status: ResultStatusTypeValidator.optional(),
  value: z4.string().optional()
}).catchall(z4.any());
var AchievementSubjectValidator = z4.object({
  id: z4.string().optional(),
  type: z4.string().array().nonempty(),
  activityEndDate: z4.string().optional(),
  activityStartDate: z4.string().optional(),
  creditsEarned: z4.number().optional(),
  achievement: AchievementValidator.optional(),
  identifier: IdentityObjectValidator.array().optional(),
  image: ImageValidator.optional(),
  licenseNumber: z4.string().optional(),
  narrative: z4.string().optional(),
  result: ResultValidator.array().optional(),
  role: z4.string().optional(),
  source: ProfileValidator.optional(),
  term: z4.string().optional()
}).catchall(z4.any());
var UnsignedAchievementCredentialValidator = UnsignedVCValidator.extend({
  name: z4.string().optional(),
  description: z4.string().optional(),
  image: ImageValidator.optional(),
  credentialSubject: AchievementSubjectValidator.or(
    AchievementSubjectValidator.array()
  ),
  endorsement: UnsignedVCValidator.array().optional()
});
var AchievementCredentialValidator = UnsignedAchievementCredentialValidator.extend({
  proof: ProofValidator.or(ProofValidator.array())
});

// src/learncard.ts
import { z as z5 } from "zod";
var VerificationCheckValidator = z5.object({
  checks: z5.string().array(),
  warnings: z5.string().array(),
  errors: z5.string().array()
});
var VerificationStatusValidator = z5.enum(["Success", "Failed", "Error"]);
var VerificationStatusEnum = VerificationStatusValidator.enum;
var VerificationItemValidator = z5.object({
  check: z5.string(),
  status: VerificationStatusValidator,
  message: z5.string().optional(),
  details: z5.string().optional()
});
var CredentialInfoValidator = z5.object({
  title: z5.string().optional(),
  createdAt: z5.string().optional(),
  issuer: ProfileValidator.optional(),
  issuee: ProfileValidator.optional(),
  credentialSubject: CredentialSubjectValidator.optional()
});
var CredentialRecordValidator = z5.object({ id: z5.string(), uri: z5.string() }).catchall(z5.any());

// src/learncloud.ts
import { z as z7 } from "zod";

// src/mongo.ts
import { z as z6 } from "zod";
var PaginationOptionsValidator = z6.object({
  limit: z6.number(),
  cursor: z6.string().optional(),
  sort: z6.string().optional()
});
var PaginationResponseValidator = z6.object({
  cursor: z6.string().optional(),
  hasMore: z6.boolean()
});

// src/learncloud.ts
var EncryptedRecordValidator = z7.object({ encryptedRecord: JWEValidator, fields: z7.string().array() }).catchall(z7.any());
var PaginatedEncryptedRecordsValidator = PaginationResponseValidator.extend({
  records: EncryptedRecordValidator.array()
});
var EncryptedCredentialRecordValidator = EncryptedRecordValidator.extend({
  id: z7.string()
});
var PaginatedEncryptedCredentialRecordsValidator = PaginationResponseValidator.extend({
  records: EncryptedCredentialRecordValidator.array()
});

// src/lcn.ts
import { z as z9 } from "zod";

// src/queries.ts
import { z as z8 } from "zod";
var parseRegexString = /* @__PURE__ */ __name((regexStr) => {
  const match = regexStr.match(/^\/(.*)\/([gimsuy]*)$/);
  if (!match) throw new Error("Invalid RegExp string format");
  return { pattern: match[1], flags: match[2] };
}, "parseRegexString");
var RegExpStringValidator = z8.string().refine(
  (str) => {
    try {
      parseRegexString(str);
      return true;
    } catch {
      return false;
    }
  },
  {
    message: "Invalid RegExp string format. Must be in format '/pattern/flags'"
  }
).transform((str) => {
  const { pattern, flags } = parseRegexString(str);
  try {
    return new RegExp(pattern, flags);
  } catch (error) {
    throw new Error(`Invalid RegExp: ${error.message}`);
  }
}).meta({ override: { type: "string" } });
var RegExpValidator = z8.instanceof(RegExp).meta({ override: { type: "string" } }).or(RegExpStringValidator).meta({ override: { type: "string" } });
var BaseStringQuery = z8.string().or(z8.object({ $in: z8.string().array() })).or(z8.object({ $regex: RegExpValidator.meta({ override: { type: "string" } }) }));
var StringQuery = z8.union([
  BaseStringQuery,
  z8.object({
    $or: BaseStringQuery.array()
  })
]);

// src/lcn.ts
var LCNProfileDisplayValidator = z9.object({
  backgroundColor: z9.string().optional(),
  backgroundImage: z9.string().optional(),
  fadeBackgroundImage: z9.boolean().optional(),
  repeatBackgroundImage: z9.boolean().optional(),
  fontColor: z9.string().optional(),
  accentColor: z9.string().optional(),
  accentFontColor: z9.string().optional(),
  idBackgroundImage: z9.string().optional(),
  fadeIdBackgroundImage: z9.boolean().optional(),
  idBackgroundColor: z9.string().optional(),
  repeatIdBackgroundImage: z9.boolean().optional()
});
var LCNProfileValidator = z9.object({
  profileId: z9.string().min(3).max(40).describe("Unique, URL-safe identifier for the profile."),
  displayName: z9.string().default("").describe("Human-readable display name for the profile."),
  shortBio: z9.string().default("").describe("Short bio for the profile."),
  bio: z9.string().default("").describe("Longer bio for the profile."),
  did: z9.string().describe("Decentralized Identifier for the profile. (auto-assigned)"),
  isPrivate: z9.boolean().optional().describe("Whether the profile is private or not and shows up in search results."),
  email: z9.string().optional().describe("Contact email address for the profile. (deprecated)"),
  image: z9.string().optional().describe("Profile image URL for the profile."),
  heroImage: z9.string().optional().describe("Hero image URL for the profile."),
  websiteLink: z9.string().optional().describe("Website link for the profile."),
  isServiceProfile: z9.boolean().default(false).optional().describe("Whether the profile is a service profile or not."),
  type: z9.string().optional().describe('Profile type: e.g. "person", "organization", "service".'),
  notificationsWebhook: z9.string().url().startsWith("http").optional().describe("URL to send notifications to."),
  display: LCNProfileDisplayValidator.optional().describe("Display settings for the profile."),
  highlightedCredentials: z9.array(z9.string()).max(5).optional().describe("Up to 5 unique boost URIs to highlight on the profile."),
  role: z9.string().default("").optional().describe('Role of the profile: e.g. "teacher", "student".'),
  dob: z9.string().default("").optional().describe('Date of birth of the profile: e.g. "1990-01-01".'),
  country: z9.string().optional().describe("Country for the profile."),
  approved: z9.boolean().optional().describe("Approval status for the profile.")
});
var LCNProfileQueryValidator = z9.object({
  profileId: StringQuery,
  displayName: StringQuery,
  shortBio: StringQuery,
  bio: StringQuery,
  email: StringQuery,
  websiteLink: StringQuery,
  isServiceProfile: z9.boolean(),
  type: StringQuery
}).partial();
var PaginatedLCNProfilesValidator = PaginationResponseValidator.extend({
  records: LCNProfileValidator.array()
});
var LCNProfileConnectionStatusEnum = z9.enum([
  "CONNECTED",
  "PENDING_REQUEST_SENT",
  "PENDING_REQUEST_RECEIVED",
  "NOT_CONNECTED"
]);
var LCNProfileManagerValidator = z9.object({
  id: z9.string(),
  created: z9.string(),
  displayName: z9.string().default("").optional(),
  shortBio: z9.string().default("").optional(),
  bio: z9.string().default("").optional(),
  email: z9.string().optional(),
  image: z9.string().optional(),
  heroImage: z9.string().optional()
});
var PaginatedLCNProfileManagersValidator = PaginationResponseValidator.extend({
  records: LCNProfileManagerValidator.extend({ did: z9.string() }).array()
});
var LCNProfileManagerQueryValidator = z9.object({
  id: StringQuery,
  displayName: StringQuery,
  shortBio: StringQuery,
  bio: StringQuery,
  email: StringQuery
}).partial();
var PaginatedLCNProfilesAndManagersValidator = PaginationResponseValidator.extend({
  records: z9.object({
    profile: LCNProfileValidator,
    manager: LCNProfileManagerValidator.extend({ did: z9.string() }).optional()
  }).array()
});
var SentCredentialInfoValidator = z9.object({
  uri: z9.string(),
  to: z9.string(),
  from: z9.string(),
  sent: z9.iso.datetime(),
  received: z9.iso.datetime().optional(),
  metadata: z9.record(z9.string(), z9.unknown()).optional()
});
var BoostPermissionsValidator = z9.object({
  role: z9.string(),
  canView: z9.boolean().default(true),
  canEdit: z9.boolean(),
  canIssue: z9.boolean(),
  canRevoke: z9.boolean(),
  canManagePermissions: z9.boolean(),
  canIssueChildren: z9.string(),
  canCreateChildren: z9.string(),
  canEditChildren: z9.string(),
  canRevokeChildren: z9.string(),
  canManageChildrenPermissions: z9.string(),
  canManageChildrenProfiles: z9.boolean().optional(),
  canViewAnalytics: z9.boolean()
});
var BoostPermissionsQueryValidator = z9.object({
  role: StringQuery,
  canView: z9.boolean(),
  canEdit: z9.boolean(),
  canIssue: z9.boolean(),
  canRevoke: z9.boolean(),
  canManagePermissions: z9.boolean(),
  canIssueChildren: StringQuery,
  canCreateChildren: StringQuery,
  canEditChildren: StringQuery,
  canRevokeChildren: StringQuery,
  canManageChildrenPermissions: StringQuery,
  canManageChildrenProfiles: z9.boolean(),
  canViewAnalytics: z9.boolean()
}).partial();
var ClaimHookTypeValidator = z9.enum(["GRANT_PERMISSIONS", "ADD_ADMIN", "AUTO_CONNECT"]);
var ClaimHookValidator = z9.discriminatedUnion("type", [
  z9.object({
    type: z9.literal(ClaimHookTypeValidator.enum.GRANT_PERMISSIONS),
    data: z9.object({
      claimUri: z9.string(),
      targetUri: z9.string(),
      permissions: BoostPermissionsValidator.partial()
    })
  }),
  z9.object({
    type: z9.literal(ClaimHookTypeValidator.enum.ADD_ADMIN),
    data: z9.object({ claimUri: z9.string(), targetUri: z9.string() })
  }),
  z9.object({
    type: z9.literal(ClaimHookTypeValidator.enum.AUTO_CONNECT),
    data: z9.object({ claimUri: z9.string(), targetUri: z9.string() })
  })
]);
var ClaimHookQueryValidator = z9.object({
  type: StringQuery,
  data: z9.object({
    claimUri: StringQuery,
    targetUri: StringQuery,
    permissions: BoostPermissionsQueryValidator.partial()
  }).partial()
});
var FullClaimHookValidator = z9.object({ id: z9.string(), createdAt: z9.string(), updatedAt: z9.string() }).and(ClaimHookValidator);
var PaginatedClaimHooksValidator = PaginationResponseValidator.extend({
  records: FullClaimHookValidator.array()
});
var LCNBoostStatus = z9.enum(["DRAFT", "PROVISIONAL", "LIVE"]);
var BoostValidator = z9.object({
  uri: z9.string(),
  name: z9.string().optional(),
  type: z9.string().optional(),
  category: z9.string().optional(),
  status: LCNBoostStatus.optional(),
  autoConnectRecipients: z9.boolean().optional(),
  meta: z9.record(z9.string(), z9.any()).optional(),
  claimPermissions: BoostPermissionsValidator.optional(),
  defaultPermissions: BoostPermissionsValidator.optional(),
  allowAnyoneToCreateChildren: z9.boolean().optional()
});
var BaseBoostQueryValidator = z9.object({
  uri: StringQuery,
  name: StringQuery,
  type: StringQuery,
  category: StringQuery,
  meta: z9.record(z9.string(), StringQuery),
  status: LCNBoostStatus.or(z9.object({ $in: LCNBoostStatus.array() })),
  autoConnectRecipients: z9.boolean()
}).partial();
var BoostQueryValidator = z9.union([
  z9.object({
    $or: BaseBoostQueryValidator.array()
  }),
  BaseBoostQueryValidator
]);
var PaginatedBoostsValidator = PaginationResponseValidator.extend({
  records: BoostValidator.array()
});
var BoostRecipientValidator = z9.object({
  to: LCNProfileValidator,
  from: z9.string(),
  received: z9.string().optional(),
  uri: z9.string().optional()
});
var PaginatedBoostRecipientsValidator = PaginationResponseValidator.extend({
  records: BoostRecipientValidator.array()
});
var BoostRecipientWithChildrenValidator = z9.object({
  to: LCNProfileValidator,
  from: z9.string(),
  received: z9.string().optional(),
  boostUris: z9.array(z9.string()),
  credentialUris: z9.array(z9.string()).optional()
});
var PaginatedBoostRecipientsWithChildrenValidator = PaginationResponseValidator.extend({
  records: BoostRecipientWithChildrenValidator.array()
});
var LCNBoostClaimLinkSigningAuthorityValidator = z9.object({
  endpoint: z9.string(),
  name: z9.string(),
  did: z9.string().optional()
});
var LCNBoostClaimLinkOptionsValidator = z9.object({
  ttlSeconds: z9.number().optional(),
  totalUses: z9.number().optional()
});
var LCNSigningAuthorityValidator = z9.object({
  endpoint: z9.string()
});
var LCNSigningAuthorityForUserValidator = z9.object({
  signingAuthority: LCNSigningAuthorityValidator,
  relationship: z9.object({
    name: z9.string().max(15).regex(/^[a-z0-9-]+$/, {
      message: "The input string must contain only lowercase letters, numbers, and hyphens."
    }),
    did: z9.string(),
    isPrimary: z9.boolean().optional()
  })
});
var AutoBoostConfigValidator = z9.object({
  boostUri: z9.string(),
  signingAuthority: z9.object({
    endpoint: z9.string(),
    name: z9.string()
  })
});
var SendBoostTemplateValidator = BoostValidator.partial().omit({ uri: true, claimPermissions: true }).extend({
  credential: VCValidator.or(UnsignedVCValidator),
  claimPermissions: BoostPermissionsValidator.partial().optional(),
  skills: z9.array(
    z9.object({
      frameworkId: z9.string(),
      id: z9.string(),
      proficiencyLevel: z9.number().optional()
    })
  ).min(1).optional()
});
var SendBrandingOptionsValidator = z9.object({
  issuerName: z9.string().optional().describe("Name of the issuing organization"),
  issuerLogoUrl: z9.string().url().optional().describe("Logo URL of the issuing organization"),
  credentialName: z9.string().optional().describe("Display name for the credential"),
  recipientName: z9.string().optional().describe("Name of the recipient for personalization")
});
var SendOptionsValidator = z9.object({
  webhookUrl: z9.string().url().optional().describe("Webhook URL to receive claim notifications"),
  suppressDelivery: z9.boolean().optional().describe("If true, returns claimUrl without sending email/SMS"),
  branding: SendBrandingOptionsValidator.optional().describe("Branding for email/SMS delivery")
});
var SendBoostInputValidator = z9.object({
  type: z9.literal("boost"),
  recipient: z9.string().describe("Profile ID, DID, email, or phone number (auto-detected)"),
  contractUri: z9.string().optional(),
  templateUri: z9.string().optional(),
  template: SendBoostTemplateValidator.optional(),
  signedCredential: VCValidator.optional(),
  options: SendOptionsValidator.optional().describe(
    "Options for email/phone recipients (Universal Inbox)"
  ),
  templateData: z9.record(z9.string(), z9.unknown()).optional(),
  integrationId: z9.string().optional().describe("Integration ID for activity tracking")
}).refine((data) => data.templateUri || data.template, {
  message: "Either templateUri or template creation data must be provided.",
  path: ["templateUri"]
});
var SendInboxResponseValidator = z9.object({
  issuanceId: z9.string(),
  status: z9.enum(["PENDING", "ISSUED", "EXPIRED", "DELIVERED", "CLAIMED"]),
  claimUrl: z9.string().url().optional().describe("Present when suppressDelivery=true")
});
var SendBoostResponseValidator = z9.object({
  type: z9.literal("boost"),
  credentialUri: z9.string(),
  uri: z9.string(),
  activityId: z9.string().describe("Links to the activity lifecycle for this issuance"),
  inbox: SendInboxResponseValidator.optional().describe(
    "Present when sent via email/phone (Universal Inbox)"
  )
});
var SendInputValidator = z9.discriminatedUnion("type", [SendBoostInputValidator]);
var SendResponseValidator = z9.discriminatedUnion("type", [SendBoostResponseValidator]);
var ConsentFlowTermsStatusValidator = z9.enum(["live", "stale", "withdrawn"]);
var ConsentFlowContractValidator = z9.object({
  read: z9.object({
    anonymize: z9.boolean().optional(),
    credentials: z9.object({
      categories: z9.record(
        z9.string(),
        z9.object({
          required: z9.boolean(),
          defaultEnabled: z9.boolean().optional()
        })
      ).default({})
    }).prefault({ categories: {} }),
    personal: z9.record(
      z9.string(),
      z9.object({ required: z9.boolean(), defaultEnabled: z9.boolean().optional() })
    ).default({})
  }).prefault({ credentials: { categories: {} }, personal: {} }),
  write: z9.object({
    credentials: z9.object({
      categories: z9.record(
        z9.string(),
        z9.object({
          required: z9.boolean(),
          defaultEnabled: z9.boolean().optional()
        })
      ).default({})
    }).prefault({ categories: {} }),
    personal: z9.record(
      z9.string(),
      z9.object({ required: z9.boolean(), defaultEnabled: z9.boolean().optional() })
    ).default({})
  }).prefault({ credentials: { categories: {} }, personal: {} })
});
var ConsentFlowContractDetailsValidator = z9.object({
  contract: ConsentFlowContractValidator,
  owner: LCNProfileValidator,
  name: z9.string(),
  subtitle: z9.string().optional(),
  description: z9.string().optional(),
  reasonForAccessing: z9.string().optional(),
  image: z9.string().optional(),
  uri: z9.string(),
  needsGuardianConsent: z9.boolean().optional(),
  redirectUrl: z9.string().optional(),
  frontDoorBoostUri: z9.string().optional(),
  createdAt: z9.string(),
  updatedAt: z9.string(),
  expiresAt: z9.string().optional(),
  autoBoosts: z9.string().array().optional(),
  writers: z9.array(LCNProfileValidator).optional()
});
var PaginatedConsentFlowContractsValidator = PaginationResponseValidator.extend({
  records: ConsentFlowContractDetailsValidator.omit({ owner: true }).array()
});
var ConsentFlowContractDataValidator = z9.object({
  credentials: z9.object({ categories: z9.record(z9.string(), z9.string().array()).default({}) }),
  personal: z9.record(z9.string(), z9.string()).default({}),
  date: z9.string()
});
var PaginatedConsentFlowDataValidator = PaginationResponseValidator.extend({
  records: ConsentFlowContractDataValidator.array()
});
var ConsentFlowContractDataForDidValidator = z9.object({
  credentials: z9.object({ category: z9.string(), uri: z9.string() }).array(),
  personal: z9.record(z9.string(), z9.string()).default({}),
  date: z9.string(),
  contractUri: z9.string()
});
var PaginatedConsentFlowDataForDidValidator = PaginationResponseValidator.extend({
  records: ConsentFlowContractDataForDidValidator.array()
});
var ConsentFlowTermValidator = z9.object({
  sharing: z9.boolean().optional(),
  shared: z9.string().array().optional(),
  shareAll: z9.boolean().optional(),
  shareUntil: z9.string().optional()
});
var ConsentFlowTermsValidator = z9.object({
  read: z9.object({
    anonymize: z9.boolean().optional(),
    credentials: z9.object({
      shareAll: z9.boolean().optional(),
      sharing: z9.boolean().optional(),
      categories: z9.record(z9.string(), ConsentFlowTermValidator).default({})
    }).prefault({ categories: {} }),
    personal: z9.record(z9.string(), z9.string()).default({})
  }).prefault({ credentials: { categories: {} }, personal: {} }),
  write: z9.object({
    credentials: z9.object({ categories: z9.record(z9.string(), z9.boolean()).default({}) }).prefault({ categories: {} }),
    personal: z9.record(z9.string(), z9.boolean()).default({})
  }).prefault({ credentials: { categories: {} }, personal: {} }),
  deniedWriters: z9.array(z9.string()).optional()
});
var PaginatedConsentFlowTermsValidator = PaginationResponseValidator.extend({
  records: z9.object({
    expiresAt: z9.string().optional(),
    oneTime: z9.boolean().optional(),
    terms: ConsentFlowTermsValidator,
    contract: ConsentFlowContractDetailsValidator,
    uri: z9.string(),
    consenter: LCNProfileValidator,
    status: ConsentFlowTermsStatusValidator
  }).array()
});
var ConsentFlowContractQueryValidator = z9.object({
  read: z9.object({
    anonymize: z9.boolean().optional(),
    credentials: z9.object({
      categories: z9.record(z9.string(), z9.object({ required: z9.boolean().optional() })).optional()
    }).optional(),
    personal: z9.record(z9.string(), z9.object({ required: z9.boolean().optional() })).optional()
  }).optional(),
  write: z9.object({
    credentials: z9.object({
      categories: z9.record(z9.string(), z9.object({ required: z9.boolean().optional() })).optional()
    }).optional(),
    personal: z9.record(z9.string(), z9.object({ required: z9.boolean().optional() })).optional()
  }).optional()
});
var ConsentFlowDataQueryValidator = z9.object({
  anonymize: z9.boolean().optional(),
  credentials: z9.object({ categories: z9.record(z9.string(), z9.boolean()).optional() }).optional(),
  personal: z9.record(z9.string(), z9.boolean()).optional()
});
var ConsentFlowDataForDidQueryValidator = z9.object({
  credentials: z9.object({ categories: z9.record(z9.string(), z9.boolean()).optional() }).optional(),
  personal: z9.record(z9.string(), z9.boolean()).optional(),
  id: StringQuery.optional()
});
var ConsentFlowTermsQueryValidator = z9.object({
  read: z9.object({
    anonymize: z9.boolean().optional(),
    credentials: z9.object({
      shareAll: z9.boolean().optional(),
      sharing: z9.boolean().optional(),
      categories: z9.record(z9.string(), ConsentFlowTermValidator.optional()).optional()
    }).optional(),
    personal: z9.record(z9.string(), z9.string()).optional()
  }).optional(),
  write: z9.object({
    credentials: z9.object({ categories: z9.record(z9.string(), z9.boolean()).optional() }).optional(),
    personal: z9.record(z9.string(), z9.boolean()).optional()
  }).optional()
});
var ConsentFlowTransactionActionValidator = z9.enum([
  "consent",
  "update",
  "sync",
  "withdraw",
  "write"
]);
var ConsentFlowTransactionsQueryValidator = z9.object({
  terms: ConsentFlowTermsQueryValidator.optional(),
  action: ConsentFlowTransactionActionValidator.or(
    ConsentFlowTransactionActionValidator.array()
  ).optional(),
  date: z9.object({ $gt: z9.string() }).or(z9.object({ $lt: z9.string() })).or(z9.object({ $eq: z9.string() })).optional(),
  expiresAt: z9.object({ $gt: z9.string() }).or(z9.object({ $lt: z9.string() })).or(z9.object({ $eq: z9.string() })).optional(),
  oneTime: z9.boolean().optional()
});
var ConsentFlowTransactionValidator = z9.object({
  expiresAt: z9.string().optional(),
  oneTime: z9.boolean().optional(),
  terms: ConsentFlowTermsValidator.optional(),
  id: z9.string(),
  action: ConsentFlowTransactionActionValidator,
  date: z9.string(),
  uris: z9.string().array().optional()
});
var PaginatedConsentFlowTransactionsValidator = PaginationResponseValidator.extend({
  records: ConsentFlowTransactionValidator.array()
});
var BaseSkillFrameworkQueryValidator = z9.object({
  id: StringQuery,
  name: StringQuery,
  description: StringQuery,
  sourceURI: StringQuery,
  status: StringQuery
}).partial();
var SkillFrameworkQueryValidator = z9.union([
  z9.object({ $or: BaseSkillFrameworkQueryValidator.array() }),
  BaseSkillFrameworkQueryValidator
]);
var ContractCredentialValidator = z9.object({
  credentialUri: z9.string(),
  termsUri: z9.string(),
  contractUri: z9.string(),
  boostUri: z9.string(),
  category: z9.string().optional(),
  date: z9.string()
});
var PaginatedContractCredentialsValidator = PaginationResponseValidator.extend({
  records: ContractCredentialValidator.array()
});
var LCNNotificationTypeEnumValidator = z9.enum([
  "CONNECTION_REQUEST",
  "CONNECTION_REQUEST_EXPIRED_INVITE",
  "CONNECTION_ACCEPTED",
  "CREDENTIAL_RECEIVED",
  "CREDENTIAL_ACCEPTED",
  "BOOST_RECEIVED",
  "BOOST_ACCEPTED",
  "PRESENTATION_REQUEST",
  "PRESENTATION_RECEIVED",
  "CONSENT_FLOW_TRANSACTION",
  "ISSUANCE_CLAIMED",
  "ISSUANCE_DELIVERED",
  "ISSUANCE_ERROR",
  "PROFILE_PARENT_APPROVED",
  "APP_LISTING_SUBMITTED",
  "APP_LISTING_APPROVED",
  "APP_LISTING_REJECTED"
]);
var LCNNotificationMessageValidator = z9.object({
  title: z9.string().optional(),
  body: z9.string().optional()
});
var LCNInboxContactMethodValidator = z9.object({
  type: z9.string(),
  value: z9.string()
});
var LCNInboxStatusEnumValidator = z9.enum([
  "PENDING",
  "ISSUED",
  "EXPIRED",
  /* DEPRECATED — use ISSUED */
  "DELIVERED",
  /* DEPRECATED — use ISSUED */
  "CLAIMED"
]);
var LCNNotificationInboxValidator = z9.object({
  issuanceId: z9.string(),
  status: LCNInboxStatusEnumValidator,
  recipient: z9.object({
    contactMethod: LCNInboxContactMethodValidator.optional(),
    learnCardId: z9.string().optional()
  }),
  timestamp: z9.iso.datetime().optional()
});
var LCNNotificationDataValidator = z9.object({
  vcUris: z9.array(z9.string()).optional(),
  vpUris: z9.array(z9.string()).optional(),
  transaction: ConsentFlowTransactionValidator.optional(),
  inbox: LCNNotificationInboxValidator.optional(),
  metadata: z9.record(z9.string(), z9.unknown()).optional()
}).loose();
var LCNNotificationValidator = z9.object({
  type: LCNNotificationTypeEnumValidator,
  to: LCNProfileValidator.partial().and(z9.object({ did: z9.string() })),
  from: LCNProfileValidator.partial().and(z9.object({ did: z9.string() })),
  message: LCNNotificationMessageValidator.optional(),
  data: LCNNotificationDataValidator.optional(),
  sent: z9.iso.datetime().optional(),
  webhookUrl: z9.string().optional()
});
var AUTH_GRANT_AUDIENCE_DOMAIN_PREFIX = "auth-grant:";
var AuthGrantValidator = z9.object({
  id: z9.string(),
  name: z9.string(),
  description: z9.string().optional(),
  challenge: z9.string().startsWith(AUTH_GRANT_AUDIENCE_DOMAIN_PREFIX).min(10, { message: "Challenge is too short" }).max(100, { message: "Challenge is too long" }),
  status: z9.enum(["revoked", "active"], {
    error: /* @__PURE__ */ __name((issue) => {
      if (issue.code === "invalid_value") return "Status must be either active or revoked";
      return "Status is required";
    }, "error")
  }),
  scope: z9.string(),
  createdAt: z9.iso.datetime({ error: "createdAt must be a valid ISO 8601 datetime string" }),
  expiresAt: z9.iso.datetime({ error: "expiresAt must be a valid ISO 8601 datetime string" }).nullish().optional()
});
var FlatAuthGrantValidator = z9.object({ id: z9.string() }).catchall(z9.any());
var AuthGrantStatusValidator = z9.enum(["active", "revoked"]);
var AuthGrantQueryValidator = z9.object({
  id: StringQuery,
  name: StringQuery,
  description: StringQuery,
  status: AuthGrantStatusValidator
}).partial();
var contactMethodBase = z9.object({
  id: z9.string(),
  isVerified: z9.boolean(),
  verifiedAt: z9.string().optional(),
  isPrimary: z9.boolean(),
  createdAt: z9.string()
});
var ContactMethodValidator = z9.discriminatedUnion("type", [
  z9.object({
    type: z9.literal("email"),
    value: z9.string().email()
  }).merge(contactMethodBase),
  z9.object({
    type: z9.literal("phone"),
    value: z9.string()
    // Can be improved with a regex later
  }).merge(contactMethodBase)
]);
var createContactMethodBase = z9.object({
  isVerified: z9.boolean().optional(),
  isPrimary: z9.boolean().optional()
});
var ContactMethodCreateValidator = z9.discriminatedUnion("type", [
  z9.object({
    type: z9.literal("email"),
    value: z9.string().email()
  }).merge(createContactMethodBase),
  z9.object({
    type: z9.literal("phone"),
    value: z9.string()
  }).merge(createContactMethodBase)
]);
var ContactMethodQueryValidator = z9.discriminatedUnion("type", [
  z9.object({
    type: z9.literal("email"),
    value: z9.string().email()
  }),
  z9.object({
    type: z9.literal("phone"),
    value: z9.string()
  })
]);
var ContactMethodVerificationRequestValidator = z9.object({
  value: z9.string(),
  type: z9.enum(["email", "phone"])
});
var ContactMethodVerificationValidator = z9.object({
  token: z9.string()
});
var SetPrimaryContactMethodValidator = z9.object({
  contactMethodId: z9.string()
});
var CreateContactMethodSessionValidator = z9.object({
  contactMethod: ContactMethodVerificationRequestValidator,
  otpChallenge: z9.string()
});
var CreateContactMethodSessionResponseValidator = z9.object({
  sessionJwt: z9.string()
});
var InboxCredentialValidator = z9.object({
  id: z9.string(),
  credential: z9.string(),
  isSigned: z9.boolean(),
  currentStatus: LCNInboxStatusEnumValidator,
  isAccepted: z9.boolean().optional(),
  expiresAt: z9.string(),
  createdAt: z9.string(),
  issuerDid: z9.string(),
  webhookUrl: z9.string().optional(),
  boostUri: z9.string().optional(),
  activityId: z9.string().optional(),
  signingAuthority: z9.object({
    endpoint: z9.string().optional(),
    name: z9.string().optional()
  }).optional()
});
var PaginatedInboxCredentialsValidator = z9.object({
  hasMore: z9.boolean(),
  records: z9.array(InboxCredentialValidator),
  cursor: z9.string().optional()
});
var InboxCredentialQueryValidator = z9.object({
  currentStatus: LCNInboxStatusEnumValidator,
  id: z9.string(),
  isSigned: z9.boolean(),
  isAccepted: z9.boolean().optional(),
  issuerDid: z9.string(),
  boostUri: z9.string()
}).partial();
var IssueInboxSigningAuthorityValidator = z9.object({
  endpoint: z9.string().url(),
  name: z9.string()
});
var IssueInboxCredentialValidator = z9.object({
  // === CORE DATA (Required) ===
  // WHAT is being sent and WHO is it for?
  recipient: ContactMethodQueryValidator.describe("The recipient of the credential"),
  // Either credential OR templateUri must be provided
  credential: VCValidator.or(VPValidator).or(UnsignedVCValidator).optional().describe(
    "The credential to issue. If not signed, the users default signing authority will be used, or the specified signing authority in the configuration."
  ),
  templateUri: z9.string().optional().describe(
    "URI of a boost template to use for issuance. The boost credential will be resolved and used. Mutually exclusive with credential field."
  ),
  // === OPTIONAL FEATURES ===
  // Add major, distinct features at the top level.
  //consentRequest: ConsentRequestValidator.optional(),
  // === PROCESS CONFIGURATION (Optional) ===
  // HOW should this issuance be handled?
  configuration: z9.object({
    signingAuthority: IssueInboxSigningAuthorityValidator.optional().describe(
      "The signing authority to use for the credential. If not provided, the users default signing authority will be used if the credential is not signed."
    ),
    webhookUrl: z9.string().url().optional().describe("The webhook URL to receive credential issuance events."),
    expiresInDays: z9.number().min(1).max(365).optional().describe("The number of days the credential will be valid for."),
    templateData: z9.record(z9.string(), z9.unknown()).optional().describe(
      "Template data to render into the boost credential template using Mustache syntax. Only used when boostUri is provided."
    ),
    // --- For User-Facing Delivery (Email/SMS) ---
    delivery: z9.object({
      suppress: z9.boolean().optional().default(false).describe(
        "Whether to suppress delivery of the credential to the recipient. If true, the email/sms will not be sent to the recipient. Useful if you would like to manually send claim link to your users."
      ),
      template: z9.object({
        id: z9.enum(["universal-inbox-claim"]).optional().describe(
          "The template ID to use for the credential delivery. If not provided, the default template will be used."
        ),
        model: z9.object({
          issuer: z9.object({
            name: z9.string().optional().describe(
              'The name of the organization (e.g., "State University").'
            ),
            logoUrl: z9.string().url().optional().describe(
              "The URL of the organization's logo."
            )
          }).optional(),
          credential: z9.object({
            name: z9.string().optional().describe(
              'The name of the credential (e.g., "Bachelor of Science").'
            ),
            type: z9.string().optional().describe(
              'The type of the credential (e.g., "degree", "certificate").'
            )
          }).optional(),
          recipient: z9.object({
            name: z9.string().optional().describe(
              'The name of the recipient (e.g., "John Doe").'
            )
          }).optional()
        }).describe(
          "The template model to use for the credential delivery. Injects via template variables into email/sms templates. If not provided, the default template will be used."
        )
      }).optional().describe(
        "The template to use for the credential delivery. If not provided, the default template will be used."
      )
    }).optional().describe(
      "Configuration for the credential delivery i.e. email or SMS. When credentials are sent to a user who has a verified email or phone associated with their account, delivery is skipped, and the credential will be sent using in-app notifications. If not provided, the default configuration will be used."
    )
  }).optional().describe(
    "Configuration for the credential issuance. If not provided, the default configuration will be used."
  )
}).refine((data) => data.credential || data.templateUri, {
  message: "Either credential or templateUri must be provided.",
  path: ["credential"]
});
var IssueInboxCredentialResponseValidator = z9.object({
  issuanceId: z9.string(),
  status: LCNInboxStatusEnumValidator,
  recipient: ContactMethodQueryValidator,
  claimUrl: z9.string().url().optional(),
  recipientDid: z9.string().optional()
});
var ClaimInboxCredentialValidator = z9.object({
  credential: VCValidator.or(VPValidator).or(UnsignedVCValidator).describe("The credential to issue."),
  configuration: z9.object({
    publishableKey: z9.string(),
    signingAuthorityName: z9.string().optional(),
    listingId: z9.string().optional(),
    listingSlug: z9.string().optional()
  }).optional()
});
var LCNDomainOrOriginValidator = z9.union([
  z9.string().regex(
    /^(?:(?:[a-zA-Z0-9-]{1,63}\.)+[a-zA-Z]{2,63}|localhost|\d{1,3}(?:\.\d{1,3}){3})(?::\d{1,5})?$/,
    {
      message: "Must be a valid domain (incl. subdomains), localhost, or IPv4, optionally with port"
    }
  ),
  z9.string().regex(
    /^https?:\/\/(?:(?:[a-zA-Z0-9-]{1,63}\.)+[a-zA-Z]{2,63}|localhost|\d{1,3}(?:\.\d{1,3}){3})(?::\d{1,5})?$/,
    { message: "Must be a valid http(s) origin" }
  )
]);
var LCNIntegrationStatusEnum = z9.enum(["setup", "active", "paused"]);
var LCNIntegrationValidator = z9.object({
  id: z9.string(),
  name: z9.string(),
  description: z9.string().optional(),
  publishableKey: z9.string(),
  whitelistedDomains: z9.array(LCNDomainOrOriginValidator).default([]),
  // Setup/onboarding status
  status: LCNIntegrationStatusEnum.default("setup"),
  guideType: z9.string().optional(),
  guideState: z9.record(z9.string(), z9.any()).optional(),
  // Timestamps
  createdAt: z9.string().optional(),
  updatedAt: z9.string().optional()
});
var LCNIntegrationCreateValidator = z9.object({
  name: z9.string(),
  description: z9.string().optional(),
  whitelistedDomains: z9.array(LCNDomainOrOriginValidator).default([]),
  guideType: z9.string().optional()
});
var LCNIntegrationUpdateValidator = z9.object({
  name: z9.string().optional(),
  description: z9.string().optional(),
  whitelistedDomains: z9.array(LCNDomainOrOriginValidator).optional(),
  rotatePublishableKey: z9.boolean().optional(),
  // Setup/onboarding updates
  status: LCNIntegrationStatusEnum.optional(),
  guideType: z9.string().optional(),
  guideState: z9.record(z9.string(), z9.any()).optional()
});
var LCNIntegrationQueryValidator = z9.object({
  id: StringQuery,
  name: StringQuery,
  description: StringQuery,
  status: StringQuery,
  guideType: StringQuery
}).partial();
var PaginatedLCNIntegrationsValidator = PaginationResponseValidator.extend({
  records: LCNIntegrationValidator.array()
});
var ClaimTokenValidator = z9.object({
  token: z9.string(),
  contactMethodId: z9.string(),
  autoVerifyContactMethod: z9.boolean().optional().default(false),
  createdAt: z9.string(),
  expiresAt: z9.string(),
  used: z9.boolean()
});
var TagValidator = z9.object({
  id: z9.string(),
  name: z9.string().min(1),
  slug: z9.string().min(1),
  createdAt: z9.string().optional(),
  updatedAt: z9.string().optional()
});
var SkillStatusEnum = z9.enum(["active", "archived"]);
var SkillValidator = z9.object({
  id: z9.string(),
  statement: z9.string(),
  description: z9.string().optional(),
  code: z9.string().optional(),
  icon: z9.string().optional(),
  type: z9.string().default("skill"),
  status: SkillStatusEnum.default("active"),
  frameworkId: z9.string().optional(),
  createdAt: z9.string().optional(),
  updatedAt: z9.string().optional()
});
var BaseSkillQueryValidator = z9.object({
  id: StringQuery,
  statement: StringQuery,
  description: StringQuery,
  code: StringQuery,
  type: StringQuery,
  status: SkillStatusEnum.or(z9.object({ $in: SkillStatusEnum.array() }))
}).partial();
var SkillQueryValidator = z9.union([
  z9.object({
    $or: BaseSkillQueryValidator.array()
  }),
  BaseSkillQueryValidator
]);
var SkillSemanticSearchInputValidator = z9.object({
  text: z9.string().min(1),
  frameworkId: z9.string().optional(),
  limit: z9.number().int().min(1).max(200).default(50)
});
var SkillSemanticSearchResultItemValidator = SkillValidator.extend({
  score: z9.number()
});
var SkillSemanticSearchResultValidator = z9.object({
  records: SkillSemanticSearchResultItemValidator.array()
});
var SkillFrameworkStatusEnum = z9.enum(["active", "archived"]);
var SkillFrameworkValidator = z9.object({
  id: z9.string(),
  name: z9.string(),
  description: z9.string().optional(),
  image: z9.string().optional(),
  sourceURI: z9.string().url().optional(),
  isPublic: z9.boolean().default(false),
  status: SkillFrameworkStatusEnum.default("active"),
  createdAt: z9.string().optional(),
  updatedAt: z9.string().optional()
});
var PaginatedSkillFrameworksValidator = PaginationResponseValidator.extend({
  records: SkillFrameworkValidator.array()
});
var SkillTreeNodeValidator = SkillValidator.extend({
  children: z9.array(z9.lazy(() => SkillTreeNodeValidator)),
  hasChildren: z9.boolean(),
  childrenCursor: z9.string().nullable().optional()
});
var PaginatedSkillTreeValidator = z9.object({
  hasMore: z9.boolean(),
  cursor: z9.string().nullable(),
  records: z9.array(SkillTreeNodeValidator)
});
var SkillTreeNodeInputValidator = z9.lazy(
  () => z9.object({
    id: z9.string().optional(),
    statement: z9.string(),
    description: z9.string().optional(),
    code: z9.string().optional(),
    icon: z9.string().optional(),
    type: z9.string().optional(),
    status: SkillStatusEnum.optional(),
    children: z9.array(SkillTreeNodeInputValidator).optional()
  })
);
var LinkProviderFrameworkInputValidator = z9.object({
  frameworkId: z9.string()
});
var CreateManagedFrameworkInputValidator = z9.object({
  id: z9.string().optional(),
  name: z9.string().min(1),
  description: z9.string().optional(),
  image: z9.string().optional(),
  sourceURI: z9.string().url().optional(),
  isPublic: z9.boolean().optional(),
  status: SkillFrameworkStatusEnum.optional(),
  skills: z9.array(SkillTreeNodeInputValidator).optional(),
  boostUris: z9.array(z9.string()).optional()
});
var UpdateFrameworkInputValidator = z9.object({
  id: z9.string(),
  name: z9.string().min(1).optional(),
  description: z9.string().optional(),
  image: z9.string().optional(),
  sourceURI: z9.string().url().optional(),
  isPublic: z9.boolean().optional(),
  status: SkillFrameworkStatusEnum.optional()
}).refine(
  (data) => data.name !== void 0 || data.description !== void 0 || data.image !== void 0 || data.sourceURI !== void 0 || data.isPublic !== void 0 || data.status !== void 0,
  {
    message: "At least one field must be provided to update",
    path: ["name"]
  }
);
var DeleteFrameworkInputValidator = z9.object({ id: z9.string() });
var CreateManagedFrameworkBatchInputValidator = z9.object({
  frameworks: z9.array(
    CreateManagedFrameworkInputValidator.extend({
      skills: z9.array(SkillTreeNodeInputValidator).optional()
    })
  ).min(1)
});
var SkillFrameworkAdminInputValidator = z9.object({
  frameworkId: z9.string(),
  profileId: z9.string()
});
var SkillFrameworkIdInputValidator = z9.object({ frameworkId: z9.string() });
var SkillFrameworkAdminsValidator = LCNProfileValidator.array();
var SyncFrameworkInputValidator = z9.object({ id: z9.string() });
var AddTagInputValidator = z9.object({
  slug: z9.string().min(1),
  name: z9.string().min(1)
});
var CreateSkillInputValidator = z9.object({
  frameworkId: z9.string(),
  skill: SkillTreeNodeInputValidator,
  parentId: z9.string().nullable().optional()
});
var UpdateSkillInputValidator = z9.object({
  frameworkId: z9.string(),
  id: z9.string(),
  statement: z9.string().optional(),
  description: z9.string().optional(),
  code: z9.string().optional(),
  icon: z9.string().optional(),
  type: z9.string().optional(),
  status: SkillStatusEnum.optional()
}).refine(
  (data) => data.statement !== void 0 || data.description !== void 0 || data.code !== void 0 || data.icon !== void 0 || data.type !== void 0 || data.status !== void 0,
  {
    message: "At least one field must be provided to update a skill",
    path: ["statement"]
  }
);
var SkillDeletionStrategyValidator = z9.enum(["recursive", "reparent"]);
var DeleteSkillInputValidator = z9.object({
  frameworkId: z9.string(),
  id: z9.string(),
  strategy: SkillDeletionStrategyValidator.optional().default("reparent")
});
var CreateSkillsBatchInputValidator = z9.object({
  frameworkId: z9.string(),
  skills: z9.array(SkillTreeNodeInputValidator).min(1),
  parentId: z9.string().nullable().optional()
});
var ReplaceSkillFrameworkSkillsInputValidator = z9.object({
  frameworkId: z9.string(),
  skills: z9.array(SkillTreeNodeInputValidator).min(0)
});
var ReplaceSkillFrameworkSkillsResultValidator = z9.object({
  created: z9.number().int().min(0),
  updated: z9.number().int().min(0),
  deleted: z9.number().int().min(0),
  unchanged: z9.number().int().min(0),
  total: z9.number().int().min(0)
});
var CountSkillsInputValidator = z9.object({
  frameworkId: z9.string(),
  skillId: z9.string().optional(),
  recursive: z9.boolean().optional().default(false),
  onlyCountCompetencies: z9.boolean().optional().default(false)
});
var CountSkillsResultValidator = z9.object({
  count: z9.number().int().min(0)
});
var GetFullSkillTreeInputValidator = z9.object({
  frameworkId: z9.string()
});
var GetFullSkillTreeResultValidator = z9.object({
  skills: z9.array(SkillTreeNodeValidator)
});
var GetSkillPathInputValidator = z9.object({
  frameworkId: z9.string(),
  skillId: z9.string()
});
var GetSkillPathResultValidator = z9.object({
  path: z9.array(SkillValidator)
});
var FrameworkWithSkillsValidator = z9.object({
  framework: SkillFrameworkValidator,
  skills: PaginatedSkillTreeValidator
});
var AppListingStatusValidator = z9.enum(["DRAFT", "PENDING_REVIEW", "LISTED", "ARCHIVED"]);
var LaunchTypeValidator = z9.enum([
  "EMBEDDED_IFRAME",
  "SECOND_SCREEN",
  "DIRECT_LINK",
  "CONSENT_REDIRECT",
  "SERVER_HEADLESS",
  "AI_TUTOR"
]);
var PromotionLevelValidator = z9.enum([
  "FEATURED_CAROUSEL",
  "CURATED_LIST",
  "STANDARD",
  "DEMOTED"
]);
var AgeRatingValidator = z9.enum(["4+", "9+", "12+", "17+"]);
var AppStoreListingValidator = z9.object({
  listing_id: z9.string(),
  slug: z9.string().optional(),
  display_name: z9.string(),
  tagline: z9.string(),
  full_description: z9.string(),
  icon_url: z9.string(),
  app_listing_status: AppListingStatusValidator,
  launch_type: LaunchTypeValidator,
  launch_config_json: z9.string(),
  category: z9.string().optional(),
  promo_video_url: z9.string().optional(),
  promotion_level: PromotionLevelValidator.optional(),
  ios_app_store_id: z9.string().optional(),
  android_app_store_id: z9.string().optional(),
  privacy_policy_url: z9.string().optional(),
  terms_url: z9.string().optional(),
  highlights: z9.array(z9.string()).optional(),
  screenshots: z9.array(z9.string()).optional(),
  hero_background_color: z9.string().optional(),
  min_age: z9.number().int().min(0).max(18).optional(),
  age_rating: AgeRatingValidator.optional()
});
var AppStoreListingCreateValidator = AppStoreListingValidator.omit({
  listing_id: true,
  app_listing_status: true,
  promotion_level: true
});
var AppStoreListingUpdateValidator = AppStoreListingValidator.partial().omit({
  listing_id: true,
  app_listing_status: true,
  promotion_level: true
});
var InstalledAppValidator = AppStoreListingValidator.extend({
  installed_at: z9.string()
});
var PaginatedAppStoreListingsValidator = PaginationResponseValidator.extend({
  records: AppStoreListingValidator.array()
});
var PaginatedInstalledAppsValidator = PaginationResponseValidator.extend({
  records: InstalledAppValidator.array()
});
var AppBoostValidator = z9.object({
  templateAlias: z9.string().min(1).max(50).regex(/^[a-z0-9-]+$/),
  boostUri: z9.string()
});
var SendCredentialEventValidator = z9.object({
  type: z9.literal("send-credential"),
  templateAlias: z9.string(),
  templateData: z9.record(z9.string(), z9.unknown()).optional()
});
var AppEventValidator = z9.discriminatedUnion("type", [SendCredentialEventValidator]);
var AppEventInputValidator = z9.object({
  listingId: z9.string(),
  event: AppEventValidator
});
var AppEventResponseValidator = z9.record(z9.string(), z9.unknown());
var CredentialActivityEventTypeValidator = z9.enum([
  "CREATED",
  "DELIVERED",
  "CLAIMED",
  "EXPIRED",
  "FAILED"
]);
var CredentialActivityRecipientTypeValidator = z9.enum(["profile", "email", "phone"]);
var CredentialActivitySourceTypeValidator = z9.enum([
  "send",
  "sendBoost",
  "sendCredential",
  "contract",
  "claim",
  "inbox",
  "claimLink",
  "acceptCredential",
  "appEvent"
]);
var CredentialActivityValidator = z9.object({
  id: z9.string(),
  activityId: z9.string(),
  eventType: CredentialActivityEventTypeValidator,
  timestamp: z9.string(),
  actorProfileId: z9.string(),
  recipientType: CredentialActivityRecipientTypeValidator,
  recipientIdentifier: z9.string(),
  boostUri: z9.string().optional(),
  credentialUri: z9.string().optional(),
  inboxCredentialId: z9.string().optional(),
  integrationId: z9.string().optional(),
  source: CredentialActivitySourceTypeValidator,
  metadata: z9.record(z9.string(), z9.unknown()).optional()
});
var CredentialActivityWithDetailsValidator = CredentialActivityValidator.extend({
  boost: z9.object({
    id: z9.string(),
    name: z9.string().optional(),
    category: z9.string().optional()
  }).optional(),
  recipientProfile: z9.object({
    profileId: z9.string(),
    displayName: z9.string().optional()
  }).optional()
});
var PaginatedCredentialActivitiesValidator = PaginationResponseValidator.extend({
  records: CredentialActivityWithDetailsValidator.array()
});
var CredentialActivityStatsValidator = z9.object({
  total: z9.number(),
  created: z9.number(),
  delivered: z9.number(),
  claimed: z9.number(),
  expired: z9.number(),
  failed: z9.number(),
  claimRate: z9.number()
});
export {
  AUTH_GRANT_AUDIENCE_DOMAIN_PREFIX,
  AchievementCredentialValidator,
  AchievementCriteriaValidator,
  AchievementSubjectValidator,
  AchievementTypeValidator,
  AchievementValidator,
  AddTagInputValidator,
  AddressValidator,
  AgeRatingValidator,
  AlignmentTargetTypeValidator,
  AlignmentValidator,
  AppBoostValidator,
  AppEventInputValidator,
  AppEventResponseValidator,
  AppEventValidator,
  AppListingStatusValidator,
  AppStoreListingCreateValidator,
  AppStoreListingUpdateValidator,
  AppStoreListingValidator,
  AuthGrantQueryValidator,
  AuthGrantStatusValidator,
  AuthGrantValidator,
  AutoBoostConfigValidator,
  BoostPermissionsQueryValidator,
  BoostPermissionsValidator,
  BoostQueryValidator,
  BoostRecipientValidator,
  BoostRecipientWithChildrenValidator,
  BoostValidator,
  ClaimHookQueryValidator,
  ClaimHookTypeValidator,
  ClaimHookValidator,
  ClaimInboxCredentialValidator,
  ClaimTokenValidator,
  ConsentFlowContractDataForDidValidator,
  ConsentFlowContractDataValidator,
  ConsentFlowContractDetailsValidator,
  ConsentFlowContractQueryValidator,
  ConsentFlowContractValidator,
  ConsentFlowDataForDidQueryValidator,
  ConsentFlowDataQueryValidator,
  ConsentFlowTermValidator,
  ConsentFlowTermsQueryValidator,
  ConsentFlowTermsStatusValidator,
  ConsentFlowTermsValidator,
  ConsentFlowTransactionActionValidator,
  ConsentFlowTransactionValidator,
  ConsentFlowTransactionsQueryValidator,
  ContactMethodCreateValidator,
  ContactMethodQueryValidator,
  ContactMethodValidator,
  ContactMethodVerificationRequestValidator,
  ContactMethodVerificationValidator,
  ContextValidator,
  ContractCredentialValidator,
  CountSkillsInputValidator,
  CountSkillsResultValidator,
  CreateContactMethodSessionResponseValidator,
  CreateContactMethodSessionValidator,
  CreateManagedFrameworkBatchInputValidator,
  CreateManagedFrameworkInputValidator,
  CreateSkillInputValidator,
  CreateSkillsBatchInputValidator,
  CredentialActivityEventTypeValidator,
  CredentialActivityRecipientTypeValidator,
  CredentialActivitySourceTypeValidator,
  CredentialActivityStatsValidator,
  CredentialActivityValidator,
  CredentialActivityWithDetailsValidator,
  CredentialInfoValidator,
  CredentialRecordValidator,
  CredentialSchemaValidator,
  CredentialStatusValidator,
  CredentialSubjectValidator,
  CriteriaValidator,
  DeleteFrameworkInputValidator,
  DeleteSkillInputValidator,
  DidDocumentValidator,
  EncryptedCredentialRecordValidator,
  EncryptedRecordValidator,
  EndorsementCredentialValidator,
  EndorsementSubjectValidator,
  EvidenceValidator,
  FlatAuthGrantValidator,
  FrameworkWithSkillsValidator,
  FullClaimHookValidator,
  GeoCoordinatesValidator,
  GetFullSkillTreeInputValidator,
  GetFullSkillTreeResultValidator,
  GetSkillPathInputValidator,
  GetSkillPathResultValidator,
  IdentifierEntryValidator,
  IdentifierTypeValidator,
  IdentityObjectValidator,
  ImageValidator,
  InboxCredentialQueryValidator,
  InboxCredentialValidator,
  InstalledAppValidator,
  IssueInboxCredentialResponseValidator,
  IssueInboxCredentialValidator,
  IssueInboxSigningAuthorityValidator,
  JWERecipientHeaderValidator,
  JWERecipientValidator,
  JWEValidator,
  JWKValidator,
  JWKWithPrivateKeyValidator,
  KnownAchievementTypeValidator,
  LCNBoostClaimLinkOptionsValidator,
  LCNBoostClaimLinkSigningAuthorityValidator,
  LCNBoostStatus,
  LCNDomainOrOriginValidator,
  LCNInboxContactMethodValidator,
  LCNInboxStatusEnumValidator,
  LCNIntegrationCreateValidator,
  LCNIntegrationQueryValidator,
  LCNIntegrationStatusEnum,
  LCNIntegrationUpdateValidator,
  LCNIntegrationValidator,
  LCNNotificationDataValidator,
  LCNNotificationInboxValidator,
  LCNNotificationMessageValidator,
  LCNNotificationTypeEnumValidator,
  LCNNotificationValidator,
  LCNProfileConnectionStatusEnum,
  LCNProfileDisplayValidator,
  LCNProfileManagerQueryValidator,
  LCNProfileManagerValidator,
  LCNProfileQueryValidator,
  LCNProfileValidator,
  LCNSigningAuthorityForUserValidator,
  LCNSigningAuthorityValidator,
  LaunchTypeValidator,
  LinkProviderFrameworkInputValidator,
  PaginatedAppStoreListingsValidator,
  PaginatedBoostRecipientsValidator,
  PaginatedBoostRecipientsWithChildrenValidator,
  PaginatedBoostsValidator,
  PaginatedClaimHooksValidator,
  PaginatedConsentFlowContractsValidator,
  PaginatedConsentFlowDataForDidValidator,
  PaginatedConsentFlowDataValidator,
  PaginatedConsentFlowTermsValidator,
  PaginatedConsentFlowTransactionsValidator,
  PaginatedContractCredentialsValidator,
  PaginatedCredentialActivitiesValidator,
  PaginatedEncryptedCredentialRecordsValidator,
  PaginatedEncryptedRecordsValidator,
  PaginatedInboxCredentialsValidator,
  PaginatedInstalledAppsValidator,
  PaginatedLCNIntegrationsValidator,
  PaginatedLCNProfileManagersValidator,
  PaginatedLCNProfilesAndManagersValidator,
  PaginatedLCNProfilesValidator,
  PaginatedSkillFrameworksValidator,
  PaginatedSkillTreeValidator,
  PaginationOptionsValidator,
  PaginationResponseValidator,
  ProfileValidator,
  PromotionLevelValidator,
  ProofValidator,
  RefreshServiceValidator,
  RegExpValidator,
  RelatedValidator,
  ReplaceSkillFrameworkSkillsInputValidator,
  ReplaceSkillFrameworkSkillsResultValidator,
  ResultDescriptionValidator,
  ResultStatusTypeValidator,
  ResultTypeValidator,
  ResultValidator,
  RubricCriterionValidator,
  SendBoostInputValidator,
  SendBoostResponseValidator,
  SendBrandingOptionsValidator,
  SendCredentialEventValidator,
  SendInboxResponseValidator,
  SendInputValidator,
  SendOptionsValidator,
  SendResponseValidator,
  SentCredentialInfoValidator,
  ServiceValidator,
  SetPrimaryContactMethodValidator,
  SkillDeletionStrategyValidator,
  SkillFrameworkAdminInputValidator,
  SkillFrameworkAdminsValidator,
  SkillFrameworkIdInputValidator,
  SkillFrameworkQueryValidator,
  SkillFrameworkStatusEnum,
  SkillFrameworkValidator,
  SkillQueryValidator,
  SkillSemanticSearchInputValidator,
  SkillSemanticSearchResultItemValidator,
  SkillSemanticSearchResultValidator,
  SkillStatusEnum,
  SkillTreeNodeInputValidator,
  SkillTreeNodeValidator,
  SkillValidator,
  StringQuery,
  SyncFrameworkInputValidator,
  TagValidator,
  TermsOfUseValidator,
  UnsignedAchievementCredentialValidator,
  UnsignedVCValidator,
  UnsignedVPValidator,
  UpdateFrameworkInputValidator,
  UpdateSkillInputValidator,
  VC2EvidenceValidator,
  VCValidator,
  VPValidator,
  VerificationCheckValidator,
  VerificationItemValidator,
  VerificationMethodValidator,
  VerificationStatusEnum,
  VerificationStatusValidator
};
