// @learncard/partner-connect@0.2.9/dist/partner-connect.esm.js
var __defProp = Object.defineProperty;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __publicField = (obj, key, value) => __defNormalProp(obj, typeof key !== "symbol" ? key + "" : key, value);
const _PartnerConnect = class _PartnerConnect {
  constructor(options) {
    __publicField(this, "hostOrigins", ["https://learncard.app"]);
    __publicField(this, "activeHostOrigin", "https://learncard.app");
    __publicField(this, "allowNativeAppOrigins", true);
    __publicField(this, "protocol", "LEARNCARD_V1");
    __publicField(this, "requestTimeout", 3e4);
    __publicField(this, "pendingRequests");
    __publicField(this, "messageListener", null);
    __publicField(this, "isInitialized", false);
    var _a;
    const hostOrigin = (options == null ? void 0 : options.hostOrigin) || _PartnerConnect.DEFAULT_HOST_ORIGIN;
    this.hostOrigins = Array.isArray(hostOrigin) ? hostOrigin : [hostOrigin];
    this.protocol = (options == null ? void 0 : options.protocol) || "LEARNCARD_V1";
    this.requestTimeout = (options == null ? void 0 : options.requestTimeout) || 3e4;
    this.allowNativeAppOrigins = (_a = options == null ? void 0 : options.allowNativeAppOrigins) != null ? _a : true;
    this.pendingRequests = /* @__PURE__ */ new Map();
    this.configureActiveOrigin();
    this.setupMessageListener();
  }
  /**
   * Configure the active host origin using the following hierarchy:
   * 1. Check for `lc_host_override` query parameter (for staging/testing)
   * 2. Fall back to first configured origin
   * 3. Fall back to DEFAULT_HOST_ORIGIN
   *
   * This origin will be used for all outgoing messages and incoming message validation.
   */
  configureActiveOrigin() {
    if (typeof window === "undefined") {
      this.activeHostOrigin = this.hostOrigins[0] || _PartnerConnect.DEFAULT_HOST_ORIGIN;
      return;
    }
    try {
      const urlParams = new URLSearchParams(window.location.search);
      const hostOverride = urlParams.get("lc_host_override");
      if (hostOverride) {
        if (this.hostOrigins.length > 0 && !this.isOriginInWhitelist(hostOverride)) {
          console.warn(
            "[LearnCard SDK] lc_host_override value is not in the configured whitelist:",
            hostOverride,
            "Allowed:",
            this.hostOrigins
          );
          this.activeHostOrigin = this.hostOrigins[0] || _PartnerConnect.DEFAULT_HOST_ORIGIN;
        } else {
          this.activeHostOrigin = hostOverride;
          console.log("[LearnCard SDK] Using lc_host_override:", hostOverride);
        }
      } else {
        this.activeHostOrigin = this.hostOrigins[0] || _PartnerConnect.DEFAULT_HOST_ORIGIN;
        console.log("[LearnCard SDK] Using configured origin:", this.activeHostOrigin);
      }
    } catch (error) {
      console.error("[LearnCard SDK] Error configuring active origin:", error);
      this.activeHostOrigin = this.hostOrigins[0] || _PartnerConnect.DEFAULT_HOST_ORIGIN;
    }
  }
  isOriginNativeApp(origin) {
    return origin.startsWith("capacitor://") || origin.startsWith("ionic://") || origin.startsWith("https://localhost") || origin.startsWith("http://localhost") || origin.startsWith("http://127.0.0.1");
  }
  /**
   * Check if an origin is in the configured whitelist
   */
  isOriginInWhitelist(origin) {
    return this.hostOrigins.includes(origin) || this.allowNativeAppOrigins && this.isOriginNativeApp(origin);
  }
  /**
   * Check if an event origin is valid against the active host origin
   *
   * Security Rule: Incoming messages must exactly match the active host origin.
   * This prevents malicious actors from spoofing origins via query parameters.
   *
   * @param eventOrigin - The origin from the MessageEvent
   * @returns true if the origin is valid
   */
  isValidOrigin(eventOrigin) {
    return eventOrigin === this.activeHostOrigin;
  }
  /**
   * Set up the central message listener to handle responses from the LearnCard host
   */
  setupMessageListener() {
    if (typeof window === "undefined") {
      throw new Error("PartnerConnect SDK can only be used in a browser environment");
    }
    this.messageListener = (event) => {
      if (!this.isValidOrigin(event.origin)) {
        return;
      }
      const data = event.data;
      if (data.protocol !== this.protocol || !data.requestId) {
        return;
      }
      const pending = this.pendingRequests.get(data.requestId);
      if (!pending) {
        return;
      }
      clearTimeout(pending.timeoutId);
      this.pendingRequests.delete(data.requestId);
      if (data.type === "SUCCESS") {
        pending.resolve(data.data);
      } else if (data.type === "ERROR") {
        pending.reject(
          data.error || { code: "UNKNOWN_ERROR", message: "An unknown error occurred" }
        );
      }
    };
    window.addEventListener("message", this.messageListener);
    this.isInitialized = true;
  }
  /**
   * Generate a unique request ID
   */
  generateRequestId(action) {
    return `${action}-${Date.now()}-${Math.random().toString(36).substring(2, 9)}`;
  }
  /**
   * Send a message to the parent window and return a Promise
   */
  sendMessage(action, payload) {
    if (!this.isInitialized) {
      return Promise.reject({
        code: "SDK_NOT_INITIALIZED",
        message: "SDK is not initialized"
      });
    }
    return new Promise((resolve, reject) => {
      const requestId = this.generateRequestId(action);
      const timeoutId = setTimeout(() => {
        if (this.pendingRequests.has(requestId)) {
          this.pendingRequests.delete(requestId);
          reject({
            code: "LC_TIMEOUT",
            message: `Request ${action} timed out after ${this.requestTimeout}ms`
          });
        }
      }, this.requestTimeout);
      this.pendingRequests.set(requestId, {
        resolve,
        reject,
        timeoutId
      });
      const message = {
        protocol: this.protocol,
        action,
        requestId,
        payload
      };
      window.parent.postMessage(message, this.activeHostOrigin);
    });
  }
  /**
   * Request user identity (Single Sign-On)
   *
   * @returns Promise resolving to user identity including DID and JWT token
   * @throws {LearnCardError} When user is not authenticated or request fails
   *
   * @example
   * ```typescript
   * const identity = await learnCard.requestIdentity();
   * console.log('User DID:', identity.user.did);
   * console.log('JWT Token:', identity.token);
   * ```
   */
  requestIdentity() {
    return this.sendMessage("REQUEST_IDENTITY", {
      challenge: `${Date.now()}-${Math.random().toString(36).substring(2, 9)}`
    });
  }
  /**
   * Send a credential to the user's LearnCard wallet
   *
   * Supports two modes:
   * 1. **Raw credential**: Pass a full verifiable credential object
   * 2. **Template-based**: Pass `{ templateAlias, templateData }` to issue from a pre-configured boost template
   *
   * @param input - Either a verifiable credential or a template-based input
   * @returns Promise resolving to credential response
   *
   * @example Raw credential
   * ```typescript
   * const response = await learnCard.sendCredential({
   *   '@context': ['https://www.w3.org/2018/credentials/v1'],
   *   type: ['VerifiableCredential'],
   *   credentialSubject: { id: 'did:example:123' }
   * });
   * console.log('Credential ID:', response.credentialId);
   * ```
   *
   * @example Template-based (for App Store apps)
   * ```typescript
   * const response = await learnCard.sendCredential({
   *   templateAlias: 'achievement-badge',
   *   templateData: { score: '95', courseName: 'Web Dev 101' }
   * });
   * console.log('Credential URI:', response.credentialUri);
   * ```
   */
  sendCredential(input) {
    if (input && typeof input === "object" && "templateAlias" in input && typeof input.templateAlias === "string") {
      const templateInput = input;
      return this.sendAppEvent({
        type: "send-credential",
        templateAlias: templateInput.templateAlias,
        templateData: templateInput.templateData
      });
    }
    return this.sendMessage("SEND_CREDENTIAL", { credential: input });
  }
  /**
   * Launch a feature in the LearnCard host application
   *
   * @param featurePath - Path to the feature (e.g., '/ai/topics')
   * @param initialPrompt - Optional initial prompt or data
   *
   * @example
   * ```typescript
   * await learnCard.launchFeature(
   *   '/ai/topics?shortCircuitStep=newTopic',
   *   'Help me understand cryptography'
   * );
   * ```
   */
  launchFeature(featurePath, initialPrompt) {
    return this.sendMessage("LAUNCH_FEATURE", { featurePath, initialPrompt });
  }
  /**
   * Request credentials from the user's wallet using a query
   *
   * @param verifiablePresentationRequest - VPR with query criteria
   * @returns Promise resolving to verifiable presentation
   *
   * @example
   * ```typescript
   * const response = await learnCard.askCredentialSearch({
   *   query: [{
   *     type: 'QueryByTitle',
   *     credentialQuery: {
   *       reason: 'We need to verify your skills',
   *       title: 'JavaScript Expert'
   *     }
   *   }],
   *   challenge: 'challenge-123',
   *   domain: window.location.hostname
   * });
   *
   * if (response.verifiablePresentation) {
   *   console.log('Received credentials:', response.verifiablePresentation.verifiableCredential);
   * }
   * ```
   */
  askCredentialSearch(verifiablePresentationRequest) {
    return this.sendMessage("ASK_CREDENTIAL_SEARCH", {
      verifiablePresentationRequest
    });
  }
  /**
   * Request a specific credential by ID
   *
   * @param credentialId - The ID of the credential to request
   * @returns Promise resolving to the credential
   *
   * @example
   * ```typescript
   * const response = await learnCard.askCredentialSpecific('credential-id-123');
   * if (response.credential) {
   *   console.log('Received credential:', response.credential);
   * }
   * ```
   */
  askCredentialSpecific(credentialId) {
    return this.sendMessage("ASK_CREDENTIAL_SPECIFIC", {
      credentialId
    });
  }
  /**
   * Request user consent for permissions
   *
   * @param contractUri - URI of the consent contract
   * @returns Promise resolving to consent response
   *
   * @example
   * ```typescript
   * // Without redirect (default) - returns VP in response if app owns the contract
   * const response = await learnCard.requestConsent('lc:network:network.learncard.com/trpc:contract:abc123');
   * if (response.granted) {
   *   console.log('User granted consent');
   *   if (response.vp) {
   *     console.log('VP:', response.vp);
   *   }
   * }
   *
   * // With redirect - redirects to contract's redirectUrl with VP in URL params
   * const response = await learnCard.requestConsent('lc:network:network.learncard.com/trpc:contract:abc123', { redirect: true });
   * ```
   */
  requestConsent(contractUri, options = {}) {
    const { redirect = false } = options;
    return this.sendMessage("REQUEST_CONSENT", {
      contractUri,
      redirect
    });
  }
  /**
   * Initiate a template-based credential issuance flow
   *
   * @param templateId - ID of the template/boost to issue
   * @param draftRecipients - Optional array of recipient DIDs
   * @returns Promise resolving to template issue response
   *
   * @example
   * ```typescript
   * const response = await learnCard.initiateTemplateIssue(
   *   'lc:network:network.learncard.com/trpc:boost:xyz789',
   *   ['did:key:abc', 'did:key:def']
   * );
   *
   * if (response.issued) {
   *   console.log('Template issued successfully');
   * }
   * ```
   */
  initiateTemplateIssue(templateId, draftRecipients) {
    return this.sendMessage("INITIATE_TEMPLATE_ISSUE", {
      templateId,
      draftRecipients: draftRecipients || []
    });
  }
  /**
   * Send a generic event to be processed by the brain service on behalf of this app.
   * This is used for backend-like operations such as issuing credentials.
   *
   * @param event - The event payload to send
   * @returns Promise resolving to the event response
   *
   * @example
   * ```typescript
   * // Issue a credential to the current user
   * const response = await learnCard.sendAppEvent({
   *   type: 'send-credential',
   *   templateAlias: 'achievement-badge',
   *   templateData: { score: '95' }
   * });
   *
   * if (response.credentialUri) {
   *   console.log('Credential issued:', response.credentialUri);
   * }
   * ```
   */
  sendAppEvent(event) {
    return this.sendMessage("APP_EVENT", event);
  }
  /**
   * Clean up the SDK and remove event listeners
   */
  destroy() {
    if (this.messageListener) {
      window.removeEventListener("message", this.messageListener);
      this.messageListener = null;
    }
    for (const [requestId, pending] of this.pendingRequests.entries()) {
      clearTimeout(pending.timeoutId);
      pending.reject({
        code: "SDK_DESTROYED",
        message: "SDK was destroyed before request completed"
      });
    }
    this.pendingRequests.clear();
    this.isInitialized = false;
  }
};
/** Default host origin (security anchor) */
__publicField(_PartnerConnect, "DEFAULT_HOST_ORIGIN", "https://learncard.app");
let PartnerConnect = _PartnerConnect;
function createPartnerConnect(options) {
  return new PartnerConnect(options);
}

export { PartnerConnect, createPartnerConnect, createPartnerConnect as default };
